src/lib/operators/get_table.cpp

```diff
@@ -42,24 +42,52 @@ std::shared_ptr<AbstractOperator> GetTable::_on_deep_copy(
 
 void GetTable::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
-std::shared_ptr<const Table> GetTable::_on_execute() {
+std::shared_ptr<const Table> GetTable::_on_execute() { Fail("GetTable can't be called without transaction context."); }
+
+std::shared_ptr<const Table> GetTable::_on_execute(std::shared_ptr<TransactionContext> transaction_context) {
   auto original_table = StorageManager::get().get_table(_name);
+
+  DebugAssert(transaction_context != nullptr || !_table_contains_discarded_chunks(original_table),
+              "Some chunks haven been discarded: They were deleted either logically (fully invalidated) or physically "
+              "(removed from memory)"
+              "Thus, GetTable needs to have a transaction context set to filter discarded chunks out.");
```

Comment:
Code style: We keep assertion texts short (so that they fit into one line) and add a code comment if necessary.